shardingsphere-distsql/shardingsphere-distsql-parser/src/main/java/org/apache/shardingsphere/distsql/parser/core/advanced/AdvancedDistSQLStatementParserEngine.java

```diff
@@ -40,12 +40,14 @@ public SQLStatement parse(final String sql) {
         ASTNode astNode = parseToASTNode(sql);
         return getSQLStatement(sql, (ParseASTNode) astNode);
     }
-    
+
     private ASTNode parseToASTNode(final String sql) {
         try {
             return SQLParserFactory.newInstance(sql, AdvancedDistSQLLexer.class, AdvancedDistSQLParser.class).parse();
-        } catch (final ParseCancellationException | SQLParsingException ignored) {
-            throw new SQLParsingException("You have an error in your SQL syntax.");
+        } catch (final ParseCancellationException ignored) {
+            throw new SQLParsingException("You have an error in your SQL syntax: " + sql);
+        } catch (final SQLParsingException sqlParsingException) {
+            throw sqlParsingException;
```

Comment:
Why separate this two exception, how about just throw together?
Does the ParseCancellationException include SQL and parse error cause?

Reply:
The ParseCancellationException does not include SQL and parse error cause.

Reply:
ParseCancellationException maybe throw from parse mode error